services/function/src/lib.rs

```diff
@@ -310,7 +319,24 @@ mod test {
                 .build(&mut table)
                 .unwrap();
 
-            Self { table, ctx }
+            async fn service_fn(v: ()) -> Result<u32, IdProductionError> {
+                Ok(100u32)
+            }
+
+            let service = ServiceBuilder::new().service_fn(service_fn).boxed();
+            let http_client_maker = resource_providers::providers::http_client_hyper::new(service);
+
+            Self {
+                table,
+                ctx,
+                http_client_maker,
+            }
+        }
+    }
+
+    impl HostClientResourceMaker for ServerWasiView {
```

Comment:
After all this work - We now have a pattern that lets us inject wasm component resources to be used by guests and all we have to do runtime side is

implement a trait 

and..